pkg/storaged/crypto-keyslots.jsx

```diff
@@ -353,64 +342,67 @@ function edit_tang_adv(client, block, key, url, adv, passphrase) {
-                                {_("Confirm removal with passphrase")}
-                            </label>
-                            <input className="form-control" type="password" hidden={val === false}
-                                   value={val} onChange={event => change(event.target.value)} />
-                        </div>
-                        <div className="form-group">
-                            <label>
-                                <input type="radio" checked={val === false}
-                                       onChange={event => change(false)} />
-                                { fmt_to_fragments(_("Force remove passphrase in $0"), <b>{key_slot}</b>) }
-                            </label>
-                            <p className="slot-warning" hidden={val !== false}>
-                                Removing a passphrase without confirmation can be dangerous.
-                            </p>
-                        </div>
-                    </div>
-                </div>
-            );
-        }
-    };
-};
+        this.onRemove = this.onRemove.bind(this);
+    }
 
-function remove_passphrase_dialog(block, key) {
-    dialog_open({
-        Title: _("Remove passphrase"),
-        Fields: [
-            RemovePassphraseField("passphrase", key, block_name(block))
-        ],
-        Action: {
-            DangerButton: true,
-            Title: _("Remove"),
-            action: function (vals) {
-                return slot_remove(block, key.slot, vals.passphrase);
-            }
-        }
-    });
+    onRemove() {
+        this.setState({ pending: true });
+        const dev = decode_filename(this.props.block.Device);
+        const cmd = ["cryptsetup", "luksKillSlot", dev, this.props.slot.toString()];
+        if (this.state.force)
+            cmd.splice(2, 0, "-q");
+
+        const spawn = cockpit.spawn(cmd, { superuser: true, err: "message" });
+        if (!this.state.force)
+            spawn.input(this.state.passphrase + "\n");
+
+        spawn.then(this.props.onClose)
+                .catch(e => this.setState({ error: e.message }))
+                .always(() => this.setState({ pending: false }));
+    }
+
+    render() {
+        return (
+            <Modal
+                title={cockpit.format(_("Remove passphrase in key slot $0"), this.props.slot)}
+                titleIconVariant="warning"
+                isOpen
+                onClose={this.props.onClose}
+                footer={<>
+                    {
+                        this.state.error && <ModalError dialogError={this.state.error} />
+                    }
+                    <Button key="remove" variant="danger" isLoading={this.state.pending} isDisabled={this.state.pending} onClick={this.onRemove}>{_("Remove")}</Button>
+                    <Button key="cancel" variant="link" isDisabled={this.state.pending} onClick={this.props.onClose}>{_("Cancel")}</Button>
```

Comment:
Disabling the cancel only is partial, you need also to disable Esc close if you want to be accurate.
grep this onEscapePress={() => undefined}

Reply:
This is broken all over cockpit, let's not blick this PR on this.

Reply:
Oh I forgot about this